assert/assertions.go

```diff
@@ -1879,6 +1879,7 @@ func (c *CollectT) Reset() {
 	c.errors = nil
 }
 
+// Deprecated: Copy should not be used.
```

Comment:
**Overview**
Copy was only called internally in one place, and this PR removes the call. There is no reason for an integrator of testify to call it directly, so we could deprecate and remove it.

If this change is not acceptable, I will revert it.

Reply:
Deprecating is totally fine. And we must delete the method. But my ultimate aim is to just get rid of `CollectT` that should have been private.